### PR TITLE
Reduce casts for LEAD/LAG

### PR DESCRIPTION
**Pull request by comphead**
## Which issue does this PR close?

Closes #.

## Rationale for this change
Reduce number of unnecessary casts from `i64` to `usize` and vice versa

### CODE REVIEW

File: datafusion/sqllogictest/test_files/limit.slt [hunk 1]
```diff
@@ -389,18 +389,18 @@ SELECT ROW_NUMBER() OVER (PARTITION BY t1.column1) FROM t t1, t t2, t t3;
...
-logical_plan
-Aggregate: groupBy=[[t1000.i]], aggr=[[]]
---TableScan: t1000 projection=[i]
-physical_plan
-AggregateExec: mode=FinalPartitioned, gby=[i@0 as i], aggr=[]
---CoalesceBatchesExec: target_batch_size=8192
-----RepartitionExec: partitioning=Hash([i@0], 4), input_partitions=4
-------AggregateExec: mode=Partial, gby=[i@0 as i], aggr=[]
---------MemoryExec: partitions=4, partition_sizes=[1, 2, 1, 1]
+#query TT
```
Comment (comphead) [inline datafusion/sqllogictest/test_files/limit.slt:392]: This test is failing on my local, not sure if its my PR, as there is no window in the plan

File: datafusion/sqllogictest/test_files/limit.slt [hunk 2]
```diff
@@ -400,7 +400,7 @@ AggregateExec: mode=FinalPartitioned, gby=[i@0 as i], aggr=[]
 --CoalesceBatchesExec: target_batch_size=8192
 ----RepartitionExec: partitioning=Hash([i@0], 4), input_partitions=4
 ------AggregateExec: mode=Partial, gby=[i@0 as i], aggr=[]
---------MemoryExec: partitions=4, partition_sizes=[1, 2, 1, 1]
+--------MemoryExec: partitions=4, partition_sizes=[1, 1, 2, 1]
```
Comment (comphead) [inline datafusion/sqllogictest/test_files/limit.slt:403]: I dont see how the PR can affect the partition size distribution as the test plan doesn't have window. More likely the test is odd

File: datafusion/sqllogictest/test_files/limit.slt (hunk 1 above)
Comment (comphead) [reply datafusion/sqllogictest/test_files/limit.slt]: https://github.com/apache/arrow-datafusion/issues/9450

File: datafusion/sqllogictest/test_files/limit.slt (hunk 2 above)
Comment (comphead) [reply datafusion/sqllogictest/test_files/limit.slt]: https://github.com/apache/arrow-datafusion/issues/9450

**Review by mustafasrepo — APPROVED** [review]
Thanks @comphead. This PR is LGTM!.

**Pull request merged**